test/GeneratedInjectorDelegatorTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-di for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-di/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Di;
+
+use PHPUnit\Framework\TestCase;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Zend\Di\GeneratedInjectorDelegator;
+use Zend\Di\InjectorInterface;
+
+class GeneratedInjectorDelegatorTest extends TestCase
+{
+    public function testProvidedNamespaceIsNotAString()
+    {
+        $container = $this->prophesize(ContainerInterface::class);
+        $container->has('config')
+            ->willReturn(true)
+            ->shouldBeCalledTimes(1);
+        $container->get('config')
+            ->willReturn(['dependencies' => ['auto' => ['aot' => ['namespace' => 1]]]])
+            ->shouldBeCalledTimes(1);
+
+        $delegator = new GeneratedInjectorDelegator();
+
+        $this->expectException(ContainerExceptionInterface::class);
```

Comment:
Testing here just for `ContainerExceptionInterface`, using PHPUnit methods we can't check two types, we can do it using custom `try...catch`, but not sure if it is needed ?

Reply:
If you make this exception a named class, as requested above, you can check for it here and this wil no longer be an issue.